aioseop_class.php

```diff
@@ -4301,153 +4305,134 @@ function apply_description_format( $description, $post = null ) {
 	}
 
 	/**
-	 * Get Robots Meta
+	 * Gets the robots meta tag string.
 	 *
 	 * @since 2.3.5
-	 * @since 2.3.11.5 Added no index API filter hook for password protected posts.
+	 * @since 2.3.11.5 Added noindex API filter hook for password protected posts.
+	 * @since 3.2.0 Refactored function to fix various bugs.
 	 *
 	 * @return string
 	 */
 	function get_robots_meta() {
 		global $aioseop_options;
-		$opts        = $this->meta_opts;
-		$page        = $this->get_page_number();
-		$robots_meta = $tax_noindex = '';
+		$opts           = $this->meta_opts;
+		$page           = $this->get_page_number();
+		$post_type      = get_post_type();
+		$noindex        = false;
+		$nofollow       = false;
+		$robots_meta    = '';
+		$tax_noindex    = '';
+		$aiosp_noindex  = '';
+		$aiosp_nofollow = '';
+
+		if (
+				! empty( $opts ) &&
+				! is_date() &&
+				! is_author() &&
+				! is_search() &&
+				! is_home()
```

Comment:
Would these extra conditions restrict the base settings for noindex/nofollow from also being applied to categories, tags, and posts/pages?

Reply:
Good question. Those do not restrict categories, tags , posts and pages. They are needed for those archive pages because the code otherwise fetches the postmeta values from the first post on the archive page.